Source/Ultraviolet.Presentation/Shared/Controls/TextBox.cs

```diff
@@ -614,7 +614,10 @@ protected override void OnQueryCursor(MouseDevice device, CursorQueryRoutedEvent
         {
             if (IsMouseCaptured && IsMouseWithinEditor())
             {
-                data.Cursor = TextEditor?.Cursor.Resource.Cursor ?? data.Cursor;
+                if (TextEditor?.Cursor.Resource != null)
+                {
+                    data.Cursor = TextEditor?.Cursor.Resource.Cursor ?? data.Cursor;
```

Comment:
```suggestion
                    data.Cursor = TextEditor?.Cursor.Resource?.Cursor ?? data.Cursor;
```

As written, this change does the not-null check on `TextEditor` twice. You can avoid that and condense the fix down into a single line by removing the new conditional and adding an additional safe navigation operator here.

Reply:
That's right, thank you! I hadn't thought of that.